src/axom/slic/docs/sphinx/sections/architecture.rst

```diff
@@ -0,0 +1,440 @@
+that are encapsulated in  ``<...>``, which, Slic knows to interpret when
+assembling the log message.
+
+The list of keywords is summarized in the table below.
+
++---------------------+--------------------------------------------------------+
+|   keyword           |            Replaced With                               |
++=====================+========================================================+
+| **<TIMESTAMP>**     | A textual representation of the time a message is      |
+|                     | logged, as returned by ``std::asctime()``.             |
++---------------------+--------------------------------------------------------+
+| **<LEVEL>**         | The :ref:`logMessageLevel`, i.e., ``ERROR``,           |
+|                     | ``WARNING``, ``INFO``, or ``DEBUG``.                   |
++---------------------+--------------------------------------------------------+
+| **<MESSAGE>**       | The supplied message that is being logged.             |
++---------------------+--------------------------------------------------------+
+| **<FILE>**          | The file from where the message was emmitted.          |
++---------------------+--------------------------------------------------------+
+| **<LINE>**          | The line location where the message was emmitted.      |
++---------------------+--------------------------------------------------------+
+| **<TAG>**           | A string tag associated with a given message, e.g., for|
+|                     | filtering during post-processing, etc.                 |
++---------------------+--------------------------------------------------------+
+| **<RANK>**          | The MPI rank that emmitted the message. Only applicable|
+|                     | when the `Axom Toolkit`_ is compiled with MPI enabled  |
+|                     | and with MPI-aware :ref:`logStream` instances, such as,|
+|                     | the :ref:`SynchronizedOutputStream` and                |
+|                     | :ref:`LumberjackStream`.                               |
++---------------------+--------------------------------------------------------+
+
+These keywords can be combined in a string to specify a template for a log
+message.
+
+For example, the following code snippet, specifies that all reported log
+messages consist of the level, enclosed in brackets followed by the
+user-supplied log message.
+
+.. code-block:: c++
+
+    std::string format = "[<LEVEL>]: <MESSAGE>";
+
+To get the file and line location within the file where the message was
+emitted, the format string above could be amended with the following:
+
+.. code-block:: c++
+
+    std::string format = "[<LEVEL>]: <MESSAGE> \n\t<FILE>:<LINE>";
+
+This indicates that the in addition to the level and user-supplied, the
+resulting log messages will have an additional line consisting of the file
+and line where the message was emitted.
+
+.. _NativeLogStreams:
+
+Native Log Streams
```

Comment:
Does "Native" here mean that these are the `LogStream`s that come with slic (e.g. built-in)?